File: 709.转换成小写字母.go
/*
 * @lc app=leetcode.cn id=709 lang=golang
 *
 * [709] 转换成小写字母
 */

// @lc code=start
// 执行用时：0 ms, 在所有 Go 提交中击败了100.00%的用户
// 内存消耗：1.9 MB, 在所有 Go 提交中击败了98.64%的用户
func toLowerCase(s string) string {
	res := make([]rune, len(s))
	for idx, letter := range s {
		if letter >= 'A' && letter <= 'Z' {
			res[idx] = rune(int(letter) ^ 32)
		} else {
			res[idx] = letter
		}
	}
	return string(res)
}

// @lc code=end

File: 709.转换成小写字母.py
#
# @lc app=leetcode.cn id=709 lang=python3
#
# [709] 转换成小写字母
#

# @lc code=start
class Solution:
    def toLowerCase(self, s: str) -> str:
        # return s.lower()
        res = ["" for _ in range(len(s))]
        A_ord, Z_ord = ord("A"), ord("Z")
        for letter in s:
            curr_ord = ord(letter)
            if A_ord <= curr_ord <= Z_ord:
                res.append(chr(curr_ord ^ (32)))
            else:
                res.append(letter)
        return "".join(res)
# @lc code=end

